### DISCUSSION TITLE
How to Output the Value of Residual and Jacobian Terms at Each Time Step

### DISCUSSION THREAD

**Discussion by FanZhangchen** [idaholab/moose#23266] [Q&A General] [answered]
Hi All,

I have a question about the ComputeQpResidual and ComputeQpJacobian in the KernelValue. Is it similar to the AMATRX and RHS of UEL in Abaqus? If yes, how can I obtain the value of the Residual term (RHS) and Jacobian term (AMATRX) since I want to compare their value at each time step in MOOSE and Abaqus to compare the numerical differences of the matrix? 

Thank you and Best wishes,

zc

**Comment by GiudGiud** [comment, marked as answer]
Hello

You ll want to use the matrix and vector tag capabilities (see the parameters on each kernel), then output them using the TagVector/MatrixAux
https://mooseframework.inl.gov/framework_development/tagging.html
https://mooseframework.inl.gov/source/auxkernels/TagVectorAux.html
Guillaume

**Reply by hugary1995** [reply]
Be aware that TagMatrixAux only outputs the diagonal entries. I'd recommend using petsc printing routines instead. Refer to the petsc manual please.

**Reply by FanZhangchen** [reply]
Thanks for both of your advice. I will try them all.
